test/evaluation/filterShouldExistIfPredicateEvaluationHandler.spec.ts

```diff
@@ -0,0 +1,34 @@
+import fs from 'fs';
+
+import { PresentationDefinition } from '@sphereon/pe-models';
+
+import { Status } from "../../lib";
+import { EvaluationHandler } from "../../lib/evaluation/evaluationHandler";
+import { FilterShouldExistIfPredicateExistsEvaluationHandler } from "../../lib/evaluation/filterShouldExistIfPredicateExistsEvaluationHandler";
+import { HandlerCheckResult } from "../../lib/evaluation/handlerCheckResult";
+
+function getFile(path: string) {
+  return JSON.parse(fs.readFileSync(path, 'utf-8'));
+}
+
+describe('evaluate', () => {
+
+  it('should report error if the predicate is present and filter is not', function () {
+    const pdSchema: PresentationDefinition = getFile('./test/dif_pe_examples/pd/pd-simple-schema-age-predicate.json').presentation_definition;
+    delete pdSchema.input_descriptors[0].constraints.fields[0].filter;
+    const vpSimple = getFile('./test/dif_pe_examples/vp/vp-simple-age-predicate.json');
+    const evaluationHandler: EvaluationHandler = new FilterShouldExistIfPredicateExistsEvaluationHandler();
+    const results: HandlerCheckResult[] = [];
+    evaluationHandler.handle(pdSchema, vpSimple, results);
+    expect(results[0]).toEqual(new HandlerCheckResult('$.input_descriptors[0].constraints.fields[0]', '', 'FilterShouldExistIfPredicateExists', Status.ERROR, "if in the field we have predicate value, the filter value should be present as well."));
+  });
```

Comment:
This is not as per sepecification.
We do not raise an error when there is predicate present but filter missing.

Reply:
not removed yet.

Reply:
so should I delete the  `filterShouldExistIfPredicateEvaluationHandler.ts` compeletely?

Reply:
deleted